### DISCUSSION TITLE
CZML wall minimum and maximum heights

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#5260] [Q&A]
Hi all.

I tested CZML wall with minimum and maximum heights but I couldn't make it work. I could make it work with Cesium SandCastle with the following code.

`var viewer = new Cesium.Viewer("cesiumContainer");

var redWall = viewer.entities.add({
  name: "Red wall at height",
  wall: {
    positions: Cesium.Cartesian3.fromDegreesArrayHeights([
138.16897642, 35.91666667, 2242.69031, 
                        138.1689762, 35.91666644, 2242.69031, 
                        138.16891278, 35.91660117, 2242.69031, 
                        138.16883222, 35.91661756, 2243.8877, 
                        138.16868625, 35.91666644, 2244.28735, 
                        138.16868555, 35.91666667, 2244.28735, 
                        138.16865472, 35.916677, 2241.9857199999997, 
                        138.16851834, 35.91672867, 2241.78577, 
                        138.16846361, 35.91675284, 2241.78577, 
                        138.16842722, 35.916777, 2240.89063, 
                        138.1683975, 35.91680644, 2239.5906999999997, 
                        138.16835667, 35.91686756, 2239.5906999999997, 
                        138.16831111, 35.91697505, 2237.78979
    ]),
    minimumHeights: [2000,2000,2000,2000,2000,2000,1500,1500,1500,
                     1500,1500,1500,1500],
    material: Cesium.Color.RED,
  },
});

viewer.zoomTo(viewer.entities);`

Does someone knows if the minimum and maximum heights on the WALL works with Terria V8?

About CZML WALL: https://github.com/AnalyticalGraphicsInc/czml-writer/wiki/Wall

Thank you!!

Hiroo

**Comment by aminevsaziz** [comment]
I'm looking for answer too maybe some devs can help like: @na9da @peterhassall @axman6 @Layoric

**Reply by Layoric** [reply]
I had limited involvement with terriajs back in 2015, from memory @kring did a lot of the engineering on the project.

**Reply by aminevsaziz** [reply]
Too bad 🙁. I still didn't figure out how to implement cesium scripts like the one above in existing terria project. I hope @kring can provide help about the issue.

**Comment by kring** [comment]
Add CesiumJS _code_ to a TerriaJS application is a bit of work, but adding CZML is easy. Just create a file with a .czml extension and drag it into the map, or add it to the catalog with `"type": "czml"`. Here's the code in the original post written as CZML:

```
[
  {
    "id": "document",
    "version": "1.0"
  },
  {
    "name": "Red wall at height",
    "wall": {
      "positions": {
        "cartographicDegrees": [
          138.16897642, 35.91666667, 2242.69031,
          138.1689762, 35.91666644, 2242.69031,
          138.16891278, 35.91660117, 2242.69031,
          138.16883222, 35.91661756, 2243.8877,
          138.16868625, 35.91666644, 2244.28735,
          138.16868555, 35.91666667, 2244.28735,
          138.16865472, 35.916677, 2241.9857199999997,
          138.16851834, 35.91672867, 2241.78577,
          138.16846361, 35.91675284, 2241.78577,
          138.16842722, 35.916777, 2240.89063,
          138.1683975, 35.91680644, 2239.5906999999997,
          138.16835667, 35.91686756, 2239.5906999999997,
          138.16831111, 35.91697505, 2237.78979
        ]
      },
      "minimumHeights": {
        "array": [2000,2000,2000,2000,2000,2000,1500,1500,1500,1500,1500,1500,1500]
      },
      "material": {
        "solidColor": {
          "color": {
            "rgba": [255, 0, 0, 255]
          }
        }
      }
    }
  }
]
```

**Comment by na9da** [comment]
Yes you can use the czml that @kring posted.

Here is an example of embedding that in terria catalog.

https://gist.github.com/na9da/4274c7346429cd9a823df0da134e7771#file-czml-json-L1-L87
